### PR TITLE
added jv-hw-logger solution

### PR DESCRIPTION
**Pull request by Basilius-Third**

### CODE REVIEW

File: src/main/java/mate/academy/Main.java [hunk 1]
```diff
@@ -6,15 +6,19 @@
...
+    private static final Logger logger = LogManager.getLogger(Main.class);
+
     public static void main(String[] args) {
         AuthenticationService authenticationService = new AuthenticationServiceImpl();
         User user;
         try {
             user = authenticationService.login("bob", "1234");
         } catch (AuthenticationException e) {
-            e.printStackTrace();
+            logger.error("Could not authenticate user!");
```
Comment (Tamara-Paskalova) [inline src/main/java/mate/academy/Main.java:21]:
```suggestion
            logger.error("Could not authenticate user!", e);
```

File: src/main/java/mate/academy/service/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -2,11 +2,15 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Logger logger = LogManager.getLogger(AuthenticationServiceImpl.class);
+
     @Override
     public User login(String login, String password) throws AuthenticationException {
-        //TODO: add corresponding log message about method login was called
+        logger.info("Method login was called. Login - " + login);
```
Comment (Tamara-Paskalova) [inline src/main/java/mate/academy/service/AuthenticationServiceImpl.java:13]:
```suggestion
        logger.debug("Method login was called. Login - " + login);
```

Comment (Tamara-Paskalova) [reply src/main/java/mate/academy/service/AuthenticationServiceImpl.java:13]: here potentially we can get an error, so it's better use debug

File: src/main/java/mate/academy/service/OrderServiceImpl.java [hunk 3]
```diff
@@ -4,27 +4,27 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class OrderServiceImpl implements OrderService {
+    private static final Logger logger = LogManager.getLogger(OrderServiceImpl.class);
+
     @Override
     public Order completeOrder(Long userId) {
-        // TODO: add log message about method completeOrder was called
+        logger.info("Method completeOrder was called. User id: " + userId);
```
Comment (Tamara-Paskalova) [inline src/main/java/mate/academy/service/OrderServiceImpl.java:15]: the same here

File: src/main/resources/log4j2.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        <Logger name="mate.academy" level="info" additivity="false">
+            <AppenderRef ref="LogToFile"/>
+            <AppenderRef ref="LogToConsole"/>
+        </Logger>
+        <Root level="error">
+            <AppenderRef ref="LogToFile"/>
+            <AppenderRef ref="LogToConsole"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (Tamara-Paskalova) [inline src/main/resources/log4j2.xml:23]: add an empty line in the end of the file

File: src/main/java/mate/academy/Main.java (hunk 1 above)
Comment (Tamara-Paskalova) [reply src/main/java/mate/academy/Main.java:21]: You can add the name of user either

**Review by Tamara-Paskalova — CHANGES_REQUESTED** [review]

File: src/main/resources/log4j2.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            <PatternLayout pattern="%d{HH:mm:ss.SSS} %-5level %logger{36} %msg%n"/>
+        </Console>
+        <File name="LogToFile" fileName="logs/app.log">
+            <PatternLayout>
+                <Pattern>%d %p %c:%L %m%n</Pattern>
+            </PatternLayout>
+        </File>
+    </Appenders>
+    <Loggers>
+        <Logger name="mate.academy" level="info" additivity="false">
```
Comment (Optipist) [inline src/main/resources/log4j2.xml:14]: change info to debug to see debug messages

**Review by Optipist — CHANGES_REQUESTED** [review]

**Review by Optipist — APPROVED** [review]

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
 target/*
+logs/**
```
Comment (KseniiaMakarova) [inline .gitignore:4]:
```suggestion
logs/*
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**